src/api/app/models/staging/staging_project.rb

```diff
@@ -100,6 +100,18 @@ def unassign_managers_group(managers)
       relationships.find_by(group: managers, role: role).try(:destroy!)
     end
 
+    def current_status_reports
+      repositories.map(&:current_status_report) + repository_architectures.map(&:current_status_report)
```

Comment:
as current_status_report calls into the backend, I think you want to make sure it's done only once. So in one of the models it should be cached.

Reply:
cc @krauselukas maybe you sync with @saraycp about how to implement it. This is basically what we paired yesterday about ...

Reply:
@ChrisBr thanks, for the hint